### ISSUE TITLE
Cannot have C++ functions overloaded on __capability in hybrid ABI

### ISSUE THREAD

**Issue by RichardsonAlex** [CTSRD-CHERI/clang#161]
If we have the following:
```
void foo(struct a*) {}
void cap(struct a * __capability c) {}
```
We get `error: definition with same mangled name as another definition`

Do we want to support this in the hybrid ABI? If so we would have to include __capability in the name mangling.

**Comment by davidchisnall** [comment]
This looks like something that Khilan broken in the capability qualifier work.  We used to mangle `__capability`.  The logic in the Itanium ABI stuff in clang CodeGen probably needs fixing.

**Comment by kgudka** [comment]
Fixed in 80c3585410e1ddf6c0d2c934db058544c5e15616.
